Python/bltinmodule.c

```diff
@@ -2737,6 +2739,496 @@ builtin_issubclass_impl(PyObject *module, PyObject *cls,
+        _PyObject_ASSERT(f_ptr, PyCode_Check(f_ptr));
+        f_code = (PyCodeObject*)f_ptr;
+
+        _Py_VPYDBG("analysing code: ");
+        _Py_VPYDBGPRINT(f_code->co_name);
+        _Py_VPYDBG("\n");
+
+        size = PySequence_Fast_GET_SIZE(f_code->co_names);
+        _Py_VPYDBG("Enumerating %ld names\n", size);
+        for(Py_ssize_t i = 0; i < size; i++){
+            PyObject* name = PySequence_Fast_GET_ITEM(f_code->co_names, i); // name.rc = x
+            _Py_VPYDBG("name ");
+            _Py_VPYDBGPRINT(name);
+            _Py_VPYDBG(": ");
+
+            if(PyDict_Contains(globals, name)){
+                PyObject* value = PyDict_GetItem(globals, name); // value.rc = x + 1
+                _Py_VPYDBG("global(");
+                _Py_VPYDBGPRINT(value);
+                _Py_VPYDBG(") -> ");
+
+                _PyDict_SetKeyImmutable((PyDictObject*)globals, name);
+
+                if(!_Py_IsImmutable(value)){
+                    if(PyFunction_Check(value)){
+                        _Py_VPYDBG("function\n");
+
+                        _Py_SetImmutable(value);
+                        if(stack_push(f_stack, ((PyFunctionObject*)value)->func_code)){
+                            Py_DECREF(value); // value.rc = x
+                            stack_free(f_stack);
+                            // frontier freed by the caller
+                            return PyErr_NoMemory();
+                        }
+                    }else{
+                        _Py_VPYDBG("pushed\n");
+
+                        if(stack_push(frontier, value)){
+                            Py_DECREF(value);  // value.rc = x
+                            stack_free(f_stack);
+                            // frontier freed by the caller
+                            return PyErr_NoMemory();
+                        }
+                    }
+                }else{
+                    _Py_VPYDBG("immutable\n");
+                    Py_DECREF(value); // value.rc = x
+                }
+            }else if(PyDict_Contains(builtins, name)){
+                _Py_VPYDBG("builtin\n");
+
+                _PyDict_SetKeyImmutable((PyDictObject*)builtins, name);
+
+                PyObject* value = PyDict_GetItem(builtins, name); // value.rc = x + 1
+                if(!_Py_IsImmutable(value)){
+                    _Py_SetImmutable(value);
+                }
+
+                Py_DECREF(value); // value.rc = x
+            }else{
+                // TODO assert that it is an instance variable
+            }
+        }
+
+        size = PySequence_Fast_GET_SIZE(f_code->co_consts);
+        _Py_VPYDBG("Enumerating %ld consts\n", size);
+        for(Py_ssize_t i = 0; i < size; i++){
+            PyObject* value = PySequence_Fast_GET_ITEM(f_code->co_consts, i); // value.rc = x
+            _Py_VPYDBG("const ");
+            _Py_VPYDBGPRINT(value);
+            _Py_VPYDBG(": ");
+            if(!_Py_IsImmutable(value)){
+                Py_INCREF(value); // value.rc = x + 1
```

Comment:
As `stack_push` increments in the success cases, I think this logic needs fixing.